tests/test_std.rs

```diff
@@ -909,6 +909,21 @@ fn combinations_zero() {
     it::assert_equal((0..0).combinations(0), vec![vec![]]);
 }
 
+#[test]
+fn combinations_range_count() {
+    for n in 0..6 {
+        for k in 0..=n {
+            let len = (n - k + 1..=n).product::<usize>() / (1..=k).product::<usize>();
+            let mut it = (0..n).combinations(k);
+            for count in (0..=len).rev() {
+                assert_eq!(it.size_hint(), (count, Some(count)));
+                assert_eq!(it.clone().count(), count);
+                assert_eq!(it.next().is_none(), count == 0);
+            }
+        }
```

Comment:
Please add `let should_be_none = it.next(); assert(should_be_none.is_none())` after the loops.